src/common/class-syntax.k

```diff
@@ -30,22 +31,15 @@ syntax ClassMemberDec ::= FieldDec | ClassDec | MethodDec | InterfaceDec | SemiC
 syntax SemiColon ::= ";" [klabel('SemiColon)]
 
 //@ \subsection{ConstructorDeclarations.sdf}
-//todo optionals
-syntax ConstrHead   ::=  AnnoConstrModList TypeParams Type Id "(" Params ")" Throws [prefer, klabel('ConstrDecHead)]
-                       | AnnoConstrModList TypeParams Type Id "(" Params ")"    [klabel('ConstrDecHead)]
-                       | AnnoConstrModList Type Id "(" Params ")" Throws   [klabel('ConstrDecHead)]
-                       | AnnoConstrModList Type Id "(" Params ")"    [klabel('ConstrDecHead)]
+syntax ConstrHead   ::=  AnnoConstrModList OptionalTypeParams Type
+                        Id "(" Params ")" OptionalThrows [klabel('ConstrDecHead)]
```

Comment:
Hmm... you could leave a bit more indentation on the second row.
But this is just an opinion. I think @dwightguth wrote some standard on how to write K.